metals/src/main/scala/scala/meta/internal/metals/DefinitionProvider.scala

```diff
@@ -61,17 +63,28 @@ final class DefinitionProvider(
       token: CancelToken
   ): Future[DefinitionResult] = {
     val fromSemanticdb =
-      semanticdbs.textDocument(path).documentIncludingStale match {
-        case Some(doc) =>
-          definitionFromSnapshot(path, params, doc)
-        case _ =>
+      semanticdbs.textDocument(path).documentIncludingStale
+    val fromSnapshot = fromSemanticdb match {
+      case Some(doc) =>
+        definitionFromSnapshot(path, params, doc)
+      case _ =>
+        DefinitionResult.empty
+    }
+    val fromIndex =
+      if (fromSnapshot.isEmpty && remote.isEnabledForPath(path)) {
+        remote.definition(params).map(_.getOrElse(fromSnapshot))
+      } else {
+        Future.successful(fromSnapshot)
+      }
+    fromIndex.flatMap { result =>
+      if (result.isEmpty) {
+        compilers().definition(params, token)
+      } else {
+        if (result.isEmpty && fromSemanticdb.isEmpty) {
```

Comment:
Is ` fromSemanticdb.isEmpty` needed?  If the `result.isEmpty` is true then ` fromSemanticdb.isEmpty` must also be true.

Reply:
We can get an empty result when the cursor is over a non-symbol position even if there’s a SemanticDB file. My goal was to try and isolate the “SemanticDB world” from the “remote language server world”